st2common/st2common/services/packs.py

```diff
@@ -290,3 +293,116 @@ def delete_action_files_from_pack(pack_name, entry_point, metadata_file):
             'The action metadata file "%s" does not exists on disk.',
             action_metadata_file_path,
         )
+
+
+def _clone_content_to_destination_file(source_file, destination_file):
+    try:
+        shutil.copy(src=source_file, dst=destination_file)
+    except PermissionError:
+        LOG.error(
+            'No write permission for "%s" file',
+            destination_file,
+        )
+        msg = 'No permission to write in "%s" file' % (destination_file)
+        raise PermissionError(msg)
+    except FileNotFoundError:
+        LOG.error('No "workflows" directory at path: "%s"', destination_file)
+        msg = "Please make sure 'workflows' directory present in path: '%s'" % (
+            destination_file
+        )
+        raise FileNotFoundError(msg)
+    except Exception as e:
+        LOG.error(
+            'Could not copy to "%s" file. Exception was "%s"',
+            destination_file,
+            e,
+        )
+        msg = (
+            'Could not copy to "%s" file, please check the logs '
+            "or ask your StackStorm administrator to check and clone "
+            "the actions files manually" % (destination_file)
+        )
+        raise Exception(msg)
+
+
+def clone_action(
+    source_pack_base_path,
+    source_metadata_file,
+    source_entry_point,
+    source_runner_type,
+    dest_pack_base_path,
+    dest_pack,
+    dest_action,
+):
+    """
+    Prepares the path for entry point and metadata files for source and destination.
+    Clones the content from source action files to destination action files.
+    """
+
+    source_metadata_file_path = os.path.join(
+        source_pack_base_path, source_metadata_file
+    )
+    dest_metadata_file_name = "%s.yaml" % (dest_action)
+    dest_metadata_file_path = os.path.join(
+        dest_pack_base_path, "actions", dest_metadata_file_name
+    )
+
+    if source_entry_point:
+        source_entry_point_file_path = os.path.join(
+            source_pack_base_path, "actions", source_entry_point
+        )
+        if source_runner_type == "python-script":
+            dest_entry_point_file_name = "%s.py" % (dest_action)
+        elif source_runner_type == "orquesta":
+            dest_entry_point_file_name = "workflows/%s.yaml" % (dest_action)
+        else:
+            dest_entry_point_file_name = dest_action
```

Comment:
- This looks duplicate from the clone_action_db. Since we ran clone_action_db before, can we take the entry point from the dest_action here?
- Check if the workflows directory exist and create if it does not